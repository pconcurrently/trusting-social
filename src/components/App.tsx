import React from "react";
import styled, { ThemeProvider } from "styled-components";
import backgroundImage from "./background.jpeg";
import { Header } from '../components/Header';

import "./App.css";

const StyledApp = styled.div`
  background-image: url(${backgroundImage});
  background-size: cover;
  min-height: 100vh;
`;

const darkTheme = {
  THEME: 'dark',
  PRIMARY: 'black',
};

const lightTheme = {
  THEME: 'light',
  PRIMARY: 'red',
};

const App = () => {
  const [theme, setTheme] = React.useState(darkTheme);

  const switchTheme = React.useCallback(() => {
    if (theme.THEME === 'dark')  {
      setTheme(lightTheme);
    } else {
      setTheme(darkTheme);
    }
  }, [setTheme, theme]);

  return (
    <ThemeProvider theme={theme}>
      <StyledApp>
        <Header switchTheme={switchTheme} />
      </StyledApp>
    </ThemeProvider>
  );
}

export default App;
