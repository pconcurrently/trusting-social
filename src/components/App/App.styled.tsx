import React from 'react';
import styled from 'styled-components';
import backgroundImage from './background.jpeg';

export const StyledApp = styled.div`
  background-image: url(${backgroundImage});
  background-size: cover;
  min-height: 100vh;
`;