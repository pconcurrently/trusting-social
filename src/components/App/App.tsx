import React from 'react';
import { ThemeProvider } from 'styled-components';

import { StyledApp } from './StyledApp';
import { Header } from '../Header';
import { lightTheme, darkTheme } from '../../shared/themes';

export const App = () => {
  const [theme, setTheme] = React.useState(darkTheme);

  const switchTheme = React.useCallback(() => {
    if (theme.NAME === 'dark') {
      setTheme(lightTheme);
    } else {
      setTheme(darkTheme);
    }
  }, [setTheme, theme]);

  return (
    <ThemeProvider theme={theme}>
      <StyledApp>
        <Header switchTheme={switchTheme} />
      </StyledApp>
    </ThemeProvider>
  );
};
