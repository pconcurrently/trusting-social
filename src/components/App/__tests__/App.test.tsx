import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';

describe('App', () => {
  it('should render App', () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });

  it('should hide StyledDarkThemeOverlay', () => {
    const app = shallow(<App />);
    expect(app.find('StyledDarkThemeOverlay').length).toBe(0);
  });
});
