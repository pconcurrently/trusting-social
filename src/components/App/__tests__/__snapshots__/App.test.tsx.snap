// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App 1`] = `
"<ThemeProvider theme={{...}}>
  <styled.div>
    <styled.div />
    <Header switchTheme={[Function]} />
    <Search />
  </styled.div>
</ThemeProvider>"
`;
